sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -225,7 +226,9 @@ class StreamSuite extends StreamTest {
 
     val df = spark.readStream.format(classOf[FakeDefaultSource].getName).load()
     Seq("", "parquet").foreach { useV1Source =>
-      withSQLConf(SQLConf.USE_V1_SOURCE_LIST.key -> useV1Source) {
+      withSQLConf(
+        SQLConf.USE_V1_SOURCE_LIST.key -> useV1Source,
+        ParquetOutputFormat.PAGE_WRITE_CHECKSUM_ENABLED -> "false") {
```

Comment:
Disable `parquet.page.write-checksum.enabled`, otherwise:
```
[info] - DataFrame reuse *** FAILED *** (1 second, 802 milliseconds)
[info]   Decoded objects do not match expected objects:
[info]   expected: WrappedArray(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
[info]   actual:   WrappedArray(0, 1, 3, 4, 5, 6, 7, 8, 9, 10, 2)
[info]   assertnotnull(upcast(getcolumnbyordinal(0, LongType), LongType, - root class: "scala.Long"))
[info]   +- upcast(getcolumnbyordinal(0, LongType), LongType, - root class: "scala.Long")
[info]      +- getcolumnbyordinal(0, LongType) (QueryTest.scala:68)
```
This issue introduced by [PARQUET-1580](https://issues.apache.org/jira/browse/PARQUET-1580). cc @gszadovszky

Reply:
@wangyum, I've checked the code change of PARQUET-1580 (again) and still don't understand why it would cause such an issue. By disabling the CRC write you only achieve to not to write an optional field in the page headers. It should not impact any kind of ordering. If it really does it means that this ordering relies on some parameters that it shouldn't. It also means that any other potential change in the file metadata might impact this ordering.
Maybe I'm overlooking something in our code base so any comment is welcomed but if not I would suggest revisiting these unit tests.

Meanwhile, I am not experienced in Spark code so if you are fine with this workaround in a unit test I am not against it.